### PR TITLE
Use tokenize.open to detect file-specific encodings

### LINKED ISSUE #115: radon 1.5.0 fails on code with PEP263 encoding statement
Using:  
- Windows 7
- Python  2.7.10
- radon 1.5.0
- installed: colorama-0.3.8 flake8-3.3.0 mccabe-0.6.1 pycodestyle-2.3.1 pyflakes-1.5.0 radon-1.5.0
- RADONFILESCODING environment unset 

With any python source code file that starts with the following line:
`# -*- coding: iso-8859-1 -*-`

Executing `runradon mi ` for the specific python source code file,

radon 1.5.0 reports the following error:  "ERROR: encoding declaration in Unicode string (<unknown>, line 0)"

- Radon 1.4.2 had no problem with the exact same python source code file.
- Trying setting the RADONFILESCODING to various values (UTF-8, iso-8859-1) does not help; other encoding mismatches are simply reported.

-----

### PR DESCRIPTION
**Pull request by rubik**
Use [`tokenize.open`](https://github.com/python/cpython/blob/cb7fdf69ec9235cb358580f189089eaf575fb9df/Lib/tokenize.py#L448), which parses the coding directive (if present) and returns the rest of the file as a file object.

Would "fix" #115 and #127 on Python 3 only, because Python 2.7 does not have that function and it should be backported in radon.

### CODE REVIEW

File: radon/cli/tools.py [hunk 1]
```diff
@@ -62,8 +63,12 @@ def _open(path):
         if path == '-':
             yield sys.stdin
         else:
-            with _open_function(path, encoding=_encoding) as f:
-                yield f
+            try:
+                with tokenize.open(path) as f:
+                    yield f
+            except:
```
Comment (thebjorn) [inline radon/cli/tools.py]: This bare except will eat an awful lot of errors...  Perhaps `elif hasattr(tokenize, 'open'):` would be better?

This still doesn't solve the problem on Python < 3.2 though..

Comment (rubik) [reply radon/cli/tools.py]: For Python < 3.2 we can just copy `tokenize.open` in Radon's code. The except was meant to capture encoding issues. There are a few of them and I didn't know where to find a complete list.

Comment (thebjorn) [reply radon/cli/tools.py]: Not sure which encoding (decoding?) issues you are thinking of, but the except here probably just hides them (and any exceptions coming from the consumer of the `yield f`).  I've created PR #131 which pulls in tokenize.open from CPython latest (by itself that was not enough).

**Comment by jrbotros** [comment]
Can you cut a new release with this fix? Would appreciate it a ton!

**Comment by rubik** [comment]
@jrbotros Sure, I pushed version 2.0.3 to PyPI!

**Comment by jrbotros** [comment]
Thanks so much!

**Pull request closed without merging**